brick/dockerlib.py

```diff
@@ -24,7 +25,25 @@ def docker_run(tag, command, volumes=None, ports=None, environment=None):
     exit(subprocess.run(cmd, shell=True).returncode)
 
 
-def docker_build(tags, dockerfile_contents, pass_ssh=False, no_cache=False, secrets=None):
+def docker_build(tags, dockerfile_contents, pass_ssh=False, no_cache=False, secrets=None, dependency_paths=None) -> str:
+    tag_to_return = tags[-1]  # Not sure why we return an argument the caller provided
+
+    import time
+    start = time.perf_counter()
+
+    # Optimization: Skip builds if the hash of dependencies did not change since the last build.
+    # Even though Buildkit is fairly fast at verifying that nothing changed, there is still a 1
+    # second overhead for each image (steps: "resolve image config for" + "load metadata for").
```

Comment:
Wow, that is a lot! 

Locally an overhead of 1.5 second explains the 175-53 second speedup, as we skip building 80 images * 1.5 second. But I guess it can be much slower on CI.